File: Beginner/playPiano/playPiano.c
#include <stdio.h>
#include <assert.h>
#include <string.h>

int main(void){
	int T,countA,countB,flag;
	char s[100];

	scanf("%d",&T);
	assert(T>=1 && T<=500);

	while(T>0){
		scanf("%s",s);
		assert(strlen(s)>=2 && strlen(s)%2==0);

		for(int i=0;i<strlen(s);i++){
			assert(s[i]=='A' || s[i]=='B');
		}

		countA = 0;
		countB = 0;
		flag = 0;

		for(int i=0;i<strlen(s);i++){
			if(s[i]=='A'){
				countA++;
			}else{
				countB++;
			}

			if(i%2!=0 && countA!=countB){
				flag = 1;
				break;
			}
		}

		if(flag==0){
			printf("yes\n");
		}else{
			printf("no\n");
		}

		T--;
	}

	return 0;
}

File: Beginner/playPiano/playPiano.java
//run as java -ea playPiano
import java.util.Scanner;

public class playPiano{
	public static void main(String[] args) {
		int T,countA,countB,flag;
		String s;
		Scanner scan = new Scanner(System.in);

		T = scan.nextInt();
		assert(T>=1 && T<=500) : "T must be 1 <= T <= 500";

		while(T>0){
			s = scan.next();
			assert(s.length()>=2 && s.length()%2==0) : "String must be at least 2 in length and be even";

			for(int i=0;i<s.length();i++){
				assert(s.charAt(i)=='A' || s.charAt(i)=='B') : "String must only have A and B characters";
			}

			countA = 0;
			countB = 0;
			flag = 0;

			for(int i=0;i<s.length();i++){
				if(s.charAt(i)=='A'){
					countA++;
				}else{
					countB++;
				}

				if(i%2!=0 && countA!=countB){
					flag = 1;
					break;
				}
			}

			if(flag==0){
				System.out.println("yes");
			}else{
				System.out.println("no");
			}

			T--;
		}
	}
}